src/api/c++/z3++.h

```diff
@@ -82,10 +83,9 @@ namespace z3 {
     /**
        \brief Exception used to sign API usage errors.
     */
-    class exception : public std::exception {
+    class exception final : public std::exception {
```

Comment:
We can either have this be `final` or it should define a virtual destructor then obey the rule of 5 by defaulting everything else.